### ISSUE TITLE
 Artifacts with DPM++ 2M SDE Karras, even when using `use_lu_lambdas`

### ISSUE THREAD

**Issue by ivanprado** [huggingface/diffusers#6295]
### Describe the bug

The results obtained using DPM++ 2M SDE Karras contain artifacts that suggest there is some bug. This happens across different XL models, although is not so visible in the base SDXL. Using Automatic1111 with the same models and the same kind of scheduler results in high-quality results, and it seems impossible to achieve the same quality using the `diffusers` equivalent.

An image generated using Automatic1111  DPM++ 2M SDE Karras with the Juggernaut XL v6 model
![adorable_infant_dpm++_2M_SDE_Karras_jugger6xl](https://github.com/huggingface/diffusers/assets/895720/2e05d037-22ce-48ef-a59a-e45d7b1962d2)

An image generated using the equivalent scheduler from `diffusers`:
![adorable_infant_dpm++_2M_SDE_Karras_jugger6xl_diffusers](https://github.com/huggingface/diffusers/assets/895720/80b9986f-891a-4e11-a7f3-6b5573bd4cc6)

The artifacts are very visible in the last image.

The following didn't help:
* Loading the model in fp32
* `use_karras_sigmas=False`
* `use_karras_sigmas=False` and `dpmsolver++`
* `euler_at_final=True` and `use_lu_lambdas=True`

I see this happening clearly at least with the following models:
* `frankjoshua/juggernautXL_version6Rundiffusion`
*  `Lykon/dreamshaper-xl-1-0`

### Reproduction

I've created a [github repo](https://github.com/ivanprado/diffusers-scheduler-problem/tree/main) to help reproducing the problem. Also, a simpler code snippet is provided below: 

```py
from diffusers import AutoPipelineForText2Image
from diffusers.schedulers import DPMSolverMultistepScheduler
import torch

model_id = "frankjoshua/juggernautXL_version6Rundiffusion"
pipeline = AutoPipelineForText2Image.from_pretrained(
    model_id, 
    torch_dtype=torch.float16,
    variant="fp16",
    use_safetensors=True,
).to("cuda")

pipeline.scheduler = DPMSolverMultistepScheduler.from_config(
        pipeline.scheduler.config, 
        use_karras_sigmas=True,
        sde_type="sde-dpmsolver++",
        euler_at_final=True,
        use_lu_lambdas=True
        )

prompt = "Adorable infant playing with a variety of colorful rattle toys."

results = pipeline(
    prompt=prompt, 
    guidance_scale=3,
    generator=torch.Generator(device="cuda").manual_seed(42),  
    num_inference_steps=25, 
    height=768, 
    width=1344)
display(results.images[0])
```

### System Info

- `diffusers` version: 0.24.0
- Platform: Linux-5.15.0-78-generic-x86_64-with-glibc2.31
- Python version: 3.10.13
- PyTorch version (GPU?): 2.1.1 (True)
- Huggingface_hub version: 0.20.1
- Transformers version: 4.36.2
- Accelerate version: 0.25.0
- xFormers version: not installed
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: no

### Who can help?

 @yiyixuxu @patrickvonplaten

**Comment by spezialspezial** [comment]
I second this. Was short of submitting an issue myself. Interestingly KDPM2AncestralDiscreteScheduler and DPMSolverSDEScheduler are not as bad as other schedulers for SDXL, both close to the original Katherine Crowson implementation. Fooocus, Comfy, Auto1111 all have crisp output using dpmpp_2m variants

**Comment by spezialspezial** [comment]
DPMSolverMultistep-Karras
![DPMSolverMultistep-Karras](https://github.com/huggingface/diffusers/assets/75758219/34690f3d-2a80-4cd7-94d6-788cda686553)

DPMSolverSDE-Karras
![DPMSolverSDE-Karras](https://github.com/huggingface/diffusers/assets/75758219/e394b8be-d265-4abc-9787-5c6d466c2b89)

Fooocus_dpmpp_2m_sde_gpu
![Fooocus_dpmpp_2m_sde_gpu](https://github.com/huggingface/diffusers/assets/75758219/68839c5f-f6e6-4787-8cc7-36bd998a5e1c)

**Comment by spezialspezial** [comment]
DPMSolverMultistep-SDE-Lula - better but still no dice
![DPMSolverMultistep-SDE-Lula](https://github.com/huggingface/diffusers/assets/75758219/1d475eb0-1301-4908-8e8b-ee7f0719f25f)

Prompt: close-up photo portrait of a soccer player
Negative: bad, blurry
Seed: 1, cuda
Guidance: 5
Steps: 40
Model: juggernautxl_version6rundiffusion

**Comment by spezialspezial** [comment]
Likely the same isssue + investigation + proposed fix
https://github.com/huggingface/diffusers/issues/5689

Any chance we could just get the original k_samplers ported as diffusers schedulers or does that come with any legal issues?

**Comment by sayakpaul** [comment]
Cc: @LuChengTHU @yiyixuxu

**Comment by mar-muel** [comment]
Hey there - Have been looking into this as well

Shouldn't the argument by `algorithm_type` instead of `sde_type`? @ivanprado

**Comment by yiyixuxu** [comment]
Here is what I got with `euler_at_final` 

```python
from diffusers import AutoPipelineForText2Image
from diffusers.schedulers import DPMSolverMultistepScheduler
import torch

model_id = "frankjoshua/juggernautXL_version6Rundiffusion"
pipeline = AutoPipelineForText2Image.from_pretrained(
    model_id, 
    torch_dtype=torch.float16,
    variant="fp16",
    use_safetensors=True,
).to("cuda")

pipeline.scheduler = DPMSolverMultistepScheduler.from_config(
        pipeline.scheduler.config, 
        algorithm_type="sde-dpmsolver++",
        euler_at_final=True,
        )

prompt = "Adorable infant playing with a variety of colorful rattle toys."

results = pipeline(
    prompt=prompt, 
    guidance_scale=3,
    generator=torch.Generator(device="cuda").manual_seed(42),  
    num_inference_steps=25, 
    height=768, 
    width=1344)
results.images[0].save("yiyi_test_9_out.png")
```
![yiyi_test_9_out](https://github.com/huggingface/diffusers/assets/12631849/589cd046-7bb4-437f-9f8e-ac06a85448cc)

**Comment by yiyixuxu** [comment]
@spezialspezial 

You can use k samplers in diffusers with https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion_k_diffusion/pipeline_stable_diffusion_k_diffusion.py

**Comment by yiyixuxu** [comment]
@ivanprado 

would you be able to share your auto1111 setting exactly as it is? 

We looked into the difference between auto1111 and diffusers before and concluded the same issue for DPM exists in auto1111/k-sampler. However, the output you provided here is a lot better than what we would generate with `use_karras_sigmas=True` in diffusers and better than `euler_at_final` or `use_lu_lambda` ,so I wonder if anything changed and would like to revisit this

**Comment by ivanprado** [comment]
@yiyixuxu I used:
```
Adorable infant playing with a variety of colorful rattle toys.
Steps: 25, Sampler: DPM++ 2M SDE Karras, CFG scale: 3, Seed: 42, Size: 1344x768, Model hash: 1fe6c7ec54, Model: Juggernaut_XL_6, Version: v1.5.1
```
<img width="1693" alt="Screenshot 2023-12-25 at 10 14 37" src="https://github.com/huggingface/diffusers/assets/895720/9acda532-97b9-473b-8d9f-1acc5a8594ad">

**Comment by aycaecemgul** [comment]
I am having the same problem, Automatic's version of schedulers greatly outperforms HF. I wanted to use DPM++ 2M SDE Karras to generated photorealistic results but it yields unrealistic results with less detail. The updates provided in old pr's does not fix this issue. Sadly this makes it less appealing to use diffusers with SD XL.

**Comment by amlarraz** [comment]
About this issue, here are some recap, thoughts, and experiments:

- automatic1111 uses the [k-diffusion](https://github.com/crowsonkb/k-diffusion) samplers.
- The [StableDiffusionKDiffusionPipeline](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion_k_diffusion/pipeline_stable_diffusion_k_diffusion.py) is created for SD1.5 and 2.1 but not for SDXL.

We thought that the k-diffusion samplers could fix the problem or, at least, reduce the quality gap.
We've created a pipeline mixing the [StableDiffusionSDXLPipeline](https://github.com/huggingface/diffusers/blob/v0.25.0/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py#L149) and [StableDiffusionKDiffusionPipeline](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion_k_diffusion/pipeline_stable_diffusion_k_diffusion.py) and the results are ok, the noise has disappeared as you can see in the image. 
The generation parameters are the same as @ivanprado shared.

![image](https://github.com/huggingface/diffusers/assets/17318832/66bca8d3-dcf5-4c37-b2ad-2878a09a7009)

**Comment by yiyixuxu** [comment]
@amlarraz 
I added it here, I will use it to debug our scheduler. feel free to give it a review :) 

https://github.com/huggingface/diffusers/pull/6447

**Comment by amlarraz** [comment]
Hey @yiyixuxu sorry for my late response. Amazing!!! I've checked your code and I've left some comments, your code and ours are mostly the same
